android/src/main/java/com/amplitude/android/Amplitude.kt

```diff
@@ -79,6 +81,33 @@ open class Amplitude(
         return this
     }
 
+    override fun processEvent(event: BaseEvent) {
+        val eventTimestamp = event.timestamp ?: System.currentTimeMillis()
+        event.timestamp = eventTimestamp
+        if (lastEventTime < eventTimestamp) {
```

Comment:
this will break the logic for `isWithinMinTimeBetweenSessions` if the event falls into the later `startNewSessionIfNeeded`, maybe we should move this to the bottom